noir/noir-repo/aztec_macros/src/transforms/functions.rs

```diff
@@ -31,6 +31,7 @@ pub fn transform_function(
     is_initializer: bool,
     insert_init_check: bool,
     is_internal: bool,
+    insert_context_creation: bool,
```

Comment:
Unrelated, but I really dislike passing multiple bools as args. What's the rustacean way of properly doing this? Using an "options" struct?

Reply:
An options struct (which potentially implements Default) and/or you could have another version of the function with the most common values for those bools as defaults. There's also the argument that stronger types should be preferred over booleans, e.g:
```rs
transform_function(
    "foo",
    StorageOpt::Defined,
    InitializerOpt::None,
    InitCheckOpt::None,
    IsInternal::Yes,
    ContextOpt::InsertContextCreation,
);
```
I find this a bit verbose though.